python/setup.py

```diff
@@ -309,25 +310,83 @@ def __init__(self, *args, **kwargs) -> None:
 if __name__ == "__main__":
     remove_unwanted_pytorch_nvcc_flags()
     generate_build_meta()
+    files_prefill, files_decode = get_instantiation_cu()
     ext_modules = []
     ext_modules.append(
         torch_cpp_ext.CUDAExtension(
             name="flashinfer._kernels",
             sources=[
-                "csrc/single_decode.cu",
-                "csrc/single_prefill.cu",
                 "csrc/cascade.cu",
                 "csrc/page.cu",
-                "csrc/batch_decode.cu",
                 "csrc/flashinfer_ops.cu",
-                "csrc/batch_prefill.cu",
                 "csrc/sampling.cu",
                 "csrc/norm.cu",
                 "csrc/rope.cu",
                 "csrc/group_gemm.cu",
                 "csrc/quantization.cu",
+            ],
+            include_dirs=[
+                str(root.resolve() / "include"),
```

Comment:
Can we save these parameters (`include_dirs`/`extra_compile_args`/etc) as variables and reuse them for all modules?

Reply:
yep!